src/config.rs

```diff
@@ -284,24 +284,45 @@ impl Config {
             return self.domain.clone();
         }
 
-        #[link(name = "c")]
-        /*unsafe*/
-        extern "C" {
-            fn gethostname(hostname: *mut u8, size: usize) -> i32;
-        }
-
         // This buffer is far larger than what most systems will ever allow, eg.
         // linux uses 64 via _SC_HOST_NAME_MAX even though POSIX says the size
         // must be _at least_ _POSIX_HOST_NAME_MAX (255), but other systems can
         // be larger, so we just use a sufficiently sized buffer so we can defer
         // a heap allocation until the last possible moment.
         let mut hostname = [0u8; 1024];
-        unsafe {
-            if gethostname(hostname.as_mut_ptr(), mem::size_of_val(&hostname)) < 0 {
+
+        #[cfg(all(target_os = "linux", target_feature = "crt-static"))]
+        {
+            use std::{fs::File, io::Read};
+            let read_bytes = match File::open("/proc/sys/kernel/hostname")
+                .and_then(|mut f| f.read(&mut hostname))
+            {
+                Ok(read_bytes) => read_bytes,
+                Err(_) => return None,
+            };
+
+            if (1..=hostname.len()).contains(&read_bytes) && hostname[read_bytes - 1] == b'\n' {
+                hostname[read_bytes - 1] = 0
+            } else {
                 return None;
             }
```

Comment:
Where did you get this implementation?

Suggest linearizing it, more like this:

```rust
let f = File::open("/proc/sys/kernel/hostname").ok()?;
let read = f.read(&mut hostname).ok()?;
```